lighthouse-core/test/config/config-helpers-test.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const helpers = require('../../config/config-helpers.js');
+const Gatherer = require('../../gather/gatherers/gatherer.js');
+const UserTimingsAudit = require('../../audits/user-timings.js');
+
+/* eslint-env jest */
+
+describe('.deepClone', () => {
+  it('should clone things deeply', () => {
+    const input = {a: {b: {c: 1}}};
+    const output = helpers.deepClone(input);
+    expect(output).not.toBe(input);
+    expect(output).toEqual(input);
+    output.a.b.c = 2;
+    expect(input.a.b.c).toEqual(1);
+  });
+});
+
+describe('.deepCloneConfigJson', () => {
+  it('should clone a config deeply', () => {
+    const TimingGatherer = new Gatherer();
+    const input = {
+      artifacts: [{id: 'Timing', gatherer: TimingGatherer}],
+      passes: [{passName: 'defaultPass', gatherers: []}],
+      audits: [{path: 'user-timings'}],
+      categories: {random: {auditRefs: [{id: 'user-timings'}]}},
+    };
+
+    const output = helpers.deepCloneConfigJson(input);
+    expect(output).not.toBe(input);
+    expect(output).toEqual(input);
+    output.artifacts[0].id = 'NewName';
+    output.passes[0].passName = 'newName';
+    output.audits[0].path = 'new-audit';
+    output.categories.random.auditRefs[0].id = 'new-audit';
+    expect(input.artifacts[0].id).toEqual('Timing');
+    expect(input.passes[0].passName).toEqual('defaultPass');
+    expect(input.audits[0].path).toEqual('user-timings');
+    expect(input.categories.random.auditRefs[0].id).toEqual('user-timings');
+  });
+
+  it('should preserve gatherer implementations in passes', () => {
```

Comment:
Nit: Perhaps it's cleaner this way and it doesn't really matter, but it seems a bit unnecessary for these to be separate cases. We could just have a "preserves implementations" test that covers all three cases as they are all testing the same function and behavior.